homeassistant/components/assist_satellite/entity.py

```diff
@@ -169,12 +169,14 @@ async def async_internal_announce(
             raise SatelliteBusyError
 
         self._is_announcing = True
+        self._set_state(AssistSatelliteState.RESPONDING)
```

Comment:
Therese changes should not be in the same PR as the esphome changes. Tests are also missing.

Reply:
Created a new PR for this: https://github.com/home-assistant/core/pull/125632